controller/contour/namespace.go

```diff
@@ -35,19 +35,16 @@ const (
 // ensureNamespace ensures the namespace for the provided name exists.
 func (r *Reconciler) ensureNamespace(ctx context.Context, name string) error {
 	ns := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: name}}
-	err := r.Client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns)
-	switch {
-	case err == nil:
-		r.Log.Info("namespace exists; skipped adding", "name", ns.Name)
-		return nil
-	case errors.IsNotFound(err):
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns); err != nil {
+		if !errors.IsNotFound(err) {
+			return fmt.Errorf("failed to get namespace %s: %v", ns.Name, err)
```

Comment:
Nit: prefer to use `%w` for errors

https://blog.golang.org/go1.13-errors

Reply:
@jpeach commit `d70ed28` changes the verb to `%w`.